scripts/create_dataset.py

```diff
@@ -46,26 +46,323 @@ def init_dataset(dataset_name, dataset_description):
+    :param input_asset: the input json asset that has been added to a dict
+    :param dataset: the created dataset that will be written to disk
+    """
+    asset = init_asset(asset_name=input_asset["name"],
+                       root_table=input_asset["rootTable"],
+                       root_column=input_asset["rootColumn"])
+
+    # add tables and the follow section to the asset
+    asset_with_tables = add_asset_tables(asset=asset, dataset=dataset)
+    asset_with_tables_and_follow = add_asset_follow(asset=asset_with_tables, dataset=dataset)
+
+    return asset_with_tables_and_follow
+
+
+def add_asset_tables(asset, dataset):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset: the asset to be created and added to the dataset with table names
+    :param dataset: the created dataset that will be written to disk
+    """
+    for table in dataset["schema"]["tables"]:
+        asset["tables"].append(
+            {
+                "name": table["name"],
+                "columns": []
+            }
+        )
+    return asset
+
+
+def add_asset_follow(asset, dataset):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset: the asset to be created and added to the dataset with relationship names
+    :param dataset: the created dataset that will be written to disk
+    """
+    for relationship in dataset["schema"]["relationships"]:
+        asset["follow"].append(relationship["name"])
+
+    return asset
+
+
+def init_asset(asset_name, root_table, root_column):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset_name: the name of the asset being  created and added to the dataset
+    :param root_table: the name of the root table for the asset
+    :param root_column: the name of the root column for the asset
+    """
+    asset = {
+        "name": asset_name,
+        "rootTable": root_table,
+        "rootColumn": root_column,
+        "tables": [],
+        "follow": []
+    }
+    return asset
+
+
+def create_table_and_relationships(input_table, input_tables):
+    """
+    Creates a table and relationships using the input tables fields and reference field.
+    :param input_table: the input table to be used to create a new table and relationships
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    table_name = input_table["name"]
+    table = init_table(table_name=table_name)
+    relationships = []
+    for input_column in input_table["columns"]:
+        column, new_relationships = create_column_and_relationships(input_column=input_column,
+                                                                    table_name=table_name,
+                                                                    input_tables=input_tables)
+        table["columns"].append(column)
```

Comment:
This assumes that `init_table` initializes the column field. Eliminating `init_table` and inlining that dictionary creation will make things safe, but I'd also propose restructuring this method to create the new lists of columns / relationships first, and then use the final list to build the table description.